dissect/target/plugins/apps/ssh/putty.py

```diff
@@ -217,20 +223,31 @@ def construct_public_key(key_type: str, iv: str) -> str:
         - https://github.com/mkorthof/reg2kh
     """
 
+    if not isinstance(key_type, str) or not isinstance(iv, str):
+        raise ValueError("Invalid key_type or iv")
+
+    key = None
+
     if key_type == "ssh-ed25519":
         x, y = iv.split(",")
         key = ECC.construct(curve="ed25519", point_x=int(x, 16), point_y=int(y, 16))
-        return key.public_key().export_key(format="OpenSSH").split()[-1]
 
     if key_type == "ecdsa-sha2-nistp256":
         _, x, y = iv.split(",")
         key = ECC.construct(curve="NIST P-256", point_x=int(x, 16), point_y=int(y, 16))
-        return key.public_key().export_key(format="OpenSSH").split()[-1]
 
     if key_type == "rsa2":
         exponent, modulus = iv.split(",")
         key = RSA.construct((int(modulus, 16), int(exponent, 16)))
-        return key.public_key().export_key(format="OpenSSH").decode("utf-8").split()[-1]
 
-    log.warning("Could not reconstruct public key: type %s not implemented.", key_type)
-    return iv
+    if not key:
+        log.warning("Could not reconstruct public key: type %s not implemented.", key_type)
+        return iv, (None, None, None)
+
+    openssh_public_key = key.public_key().export_key(format="OpenSSH")
+
+    if isinstance(openssh_public_key, bytes):
+        openssh_public_key = openssh_public_key.decode("utf-8")
+
+    fingerprints = calculate_fingerprints(b64decode(openssh_public_key.split(" ")[1]))
+    return openssh_public_key.split()[-1], fingerprints
```

Comment:
```suggestion

    key_part = openssh_public_key.split()[-1]
    fingerprints = calculate_fingerprints(b64decode(key_part))
    return key_part, fingerprints
```
You used two different splits and indices. I'd prefer `-1` as it can't cause an `IndexError`, but I'm not sure if you deliberately used `1` here for some reason.